ledger/simulation/simulator.go

```diff
@@ -205,14 +204,41 @@ func (s Simulator) evaluate(hdr bookkeeping.BlockHeader, stxns []transactions.Si
 	return vb, nil
 }
 
-func (s Simulator) simulateWithTracer(txgroup []transactions.SignedTxn, tracer logic.EvalTracer, overrides ResultEvalOverrides) (*ledgercore.ValidatedBlock, error) {
+func (s Simulator) simulateWithTracer(txgroup []transactions.SignedTxnWithAD, tracer logic.EvalTracer, overrides ResultEvalOverrides) (*ledgercore.ValidatedBlock, error) {
 	prevBlockHdr, err := s.ledger.BlockHdr(s.ledger.start)
 	if err != nil {
 		return nil, err
 	}
 	nextBlock := bookkeeping.MakeBlock(prevBlockHdr)
 	hdr := nextBlock.BlockHeader
 
+	fixAuthAddr := true
+	if fixAuthAddr {
+		// Map of rekeys for senders in the group
+		staticRekeys := make(map[basics.Address]basics.Address)
+
+		for i := range txgroup {
+			sender := txgroup[i].SignedTxn.Txn.Sender
+
+			if authAddr, ok := staticRekeys[sender]; ok {
+				// If there is a static rekey for the sender set the auth addr to that address
+				txgroup[i].SignedTxn.AuthAddr = authAddr
```

Comment:
We may want to only modify txn AuthAddrs if the transaction has a missing signature. Otherwise the signature is a bit pointless. Worth thinking about at leaset